DemoApp/Shared/StreamChatWrapper.swift

```diff
@@ -9,14 +9,18 @@ import UIKit
 import UserNotifications
 
 final class StreamChatWrapper {
-    static let shared = StreamChatWrapper()
+    static var shared = StreamChatWrapper(apiKeyString: apiKeyString)
+
+    static func replaceSharedInstance(apiKeyString: String) {
+        StreamChatWrapper.shared = StreamChatWrapper(apiKeyString: apiKeyString)
+    }
 
     /// How many times the token has been refreshed. This is mostly used
     /// to fake token refresh fails.
     var numberOfRefreshTokens = 0
 
     // This closure is called once the SDK is ready to register for remote push notifications
-    var onRemotePushRegistration: (() -> Void)?
+    static var onRemotePushRegistration: (() -> Void)?
```

Comment:
The coordinator needs this, and was using an instance of the StreamChatWrapper. To handle the push registration, this can be static, and makes the re-configuration with the new key eaiser. Otherwise, this would require a much bigger refactor on the demo app